### PR TITLE
Integration with Metamask for payments

### PR DESCRIPTION
**Pull request by EdmundMai**
opacity_storage_-_secure__anonymous__opaque_

metamask_notification_and_opacity_storage_-_secure__anonymous__opaque_

### CODE REVIEW

**Comment by MrRedPandabear** [comment]
@EdmundMai That looks neat! Even if the orange bites the purple I'm stoked to use it.

File: src/components/payment-invoice/payment-invoice-slide.tsx [hunk 1]
```diff
@@ -6,31 +6,49 @@ import ScreenDescription from "../shared/screen-description";
...
-          size="200"
-          renderAs="svg"
-          bgColor="#ffffff"
-          fgColor="#2e3854"
-          level="H"
-          color="#ffffff"
-          includeMargin="true"
-        />
-      </ScreenDescription>
+    <MetamaskButton
```
Comment (CKH4) [inline src/components/payment-invoice/payment-invoice-slide.tsx]: This should be conditional on whether it is added

File: src/contracts/opacity.json [hunk 2]
```diff
@@ -0,0 +1,366 @@
+[
```
Comment (CKH4) [inline src/contracts/opacity.json]: This is an abi right? I think the convention is to have the extension be `.abi.json`

File: src/services/metamask.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+declare global {
+  interface Window {
+    ethereum: any;
+    web3: any;
+    Web3: any;
+  }
+}
+
+const isMetamaskInstalled = window.ethereum && window.ethereum._metamask;
```
Comment (CKH4) [inline src/services/metamask.ts]: If you do this, there is no backwards compatibility with old beta versions. I believe the old versions were at `window.web3`. This has code to enable with backwards compatibility: https://medium.com/metamask/https-medium-com-metamask-breaking-change-injecting-web3-7722797916a8

**Pull request merged**